### DISCUSSION TITLE
搜索页面有PDF时开始自动下载

### DISCUSSION THREAD

**Discussion by peterpark1996** [F9y4ng/GreasyFork-Scripts#338] [问答专区(Question / Answer)]
我注意到在搜索时会排除失效的网站并用波浪线划掉来提示，可能于此有关。当我搜索界面网站为ARVIX时会自动调用IDM弹出下载
![QQ截图20240519225827](https://github.com/F9y4ng/GreasyFork-Scripts/assets/170252317/6d14e2c9-811c-44c4-9c23-ddd5fdeb53eb)

**Comment by F9y4ng** [comment]
打开浏览器控制台(console)你可以看到脚本捕获的IDM劫持的警告信息。

![screenshot_2024-05-20_13-42-21](https://github.com/F9y4ng/GreasyFork-Scripts/assets/15855068/8927b579-0a34-4e26-9d2e-74719a7d9b48)

### 原因分析
IDM的浏览器扩展会监听每个HTTP请求，并对符合下载要求的请求进行劫持操作，从而完成自动通过IDM下载。在使用脚本时，应用去除重定向功能，解析原始URL的请求会被IDM扩展劫持从而出现直接下载的问题。

你可以通过暂时禁用IDM扩展来验证此问题。

**Reply by peterpark1996** [reply]
禁用IDM扩展后其他下载链接无法调动IDM，但是搜索引擎页面还是存在中这个问题。我直接禁用了脚本的重定向功能后不在弹窗了

**Reply by F9y4ng** [reply]
禁用IDM扩展与禁用去重定向功能的作用是一样的。最新版6.42的IDM软件自身也增加了HTTP请求监听的功能，所以禁用IDM扩展仍会出现你遇到的情况。

几乎所有预读或去重定向类的脚本或扩展都会受到IDM劫持的影响。此时，只能二者选其一来使用。

**Comment by Xia0xia0Xia0** [comment]
使用Free Download Manager是不会发生劫持问题的，而且可以正常的识别下载需求。
